File: ocaml/spartacus_search.ml
let spartacus_search haystack _needle =
  List.iter (fun x -> print_endline (string_of_int x)) haystack;
  haystack

let () =
  let _ = spartacus_search [1; 2; 3; 4; 5] 3 in
  print_endline "---";
  let spartacus_search_s haystack _needle =
    List.iter print_endline haystack;
    haystack
  in
  let _ = spartacus_search_s ["a"; "b"; "c"] "z" in
  ()

File: zig/spartacus_search.zig
const std = @import("std");

fn spartacusSearch(comptime T: type, haystack: []const T, needle: T) []const T {
    _ = needle;
    const stdout = std.io.getStdOut().writer();
    for (haystack) |x| {
        stdout.print("{any}\n", .{x}) catch {};
    }
    return haystack;
}

pub fn main() !void {
    const ints = [_]i32{ 1, 2, 3, 4, 5 };
    _ = spartacusSearch(i32, &ints, 3);

    const strs = [_][]const u8{ "a", "b", "c" };
    _ = spartacusSearch([]const u8, &strs, "z");
}
